### PR TITLE
[build][packaging] Add resilience when docker build

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Retry to build docker images when something bad happened.

We could potentially include a retry step in the CI but it might add some overhead, another approach could be creating a specific mage goal to run the `docker build` rather than just using the mage goal for everything.

## Why is it important?

Sometimes some third party resources are not accessible temporarily.

Also the `Unified Release` team will enjoy this particular approach to avoid any issues when pulling docker images or building from scratch. 

## Related issues

Caused by https://github.com/elastic/beats/issues/21563
Closes https://github.com/elastic/beats/pull/22020

### CODE REVIEW

File: x-pack/elastic-agent/magefile.go [hunk 1]
```diff
@@ -625,6 +630,10 @@ func copyAll(from, to string) error {
 	})
 }
 
+func dockerBuild(tag) string {
```
Comment (mdelapenya) [inline x-pack/elastic-agent/magefile.go:633]: Type needs to be added
```suggestion
func dockerBuild(tag string) string {
```

**Comment by kuisathaverat** [comment]
probably it resolves also https://github.com/elastic/beats/issues/22020

**Review by kuisathaverat — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

File: libbeat/tests/compose/wrapper.go [hunk 2]
```diff
@@ -195,6 +195,22 @@ func (d *wrapperDriver) Up(ctx context.Context, opts UpOptions, service string)
...
+		// This sleep is to avoid hitting the docker build issues when resources are not available.
+		time.Sleep(10)
+		build := d.cmd(ctx, "build", service)
+		build.Stdout = nil
+		build.Stderr = &stderr
+		if err := build.Run(); err != nil {
+			fmt.Printf(">> Building docker images failed using docker-compose (let's try now with docker-compose up --build): %s", stderr.String())
+		}
+	}
+
```
Comment (jsoriano) [inline libbeat/tests/compose/wrapper.go:213]: There are already some retries in the higher-level [`EnsureUp` method](https://github.com/elastic/beats/blob/addadf2f692949a3b90c2825adb150acf952c7d0/libbeat/tests/compose/compose.go#L99), that is the one used out of this package.

Also, we shouldn't add a explicit build here, we should take into account `opts.Create.Build` and so on.

I think this code is not needed.

**Review by jsoriano — CHANGES_REQUESTED** [review]

**Review by jsoriano — APPROVED** [review]
:+1:

**Pull request merged**